src/main/java/net/minestom/server/lock/Acquisition.java

```diff
@@ -0,0 +1,174 @@
+package net.minestom.server.lock;
+
+import net.minestom.server.MinecraftServer;
+import net.minestom.server.thread.BatchThread;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.*;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.concurrent.locks.ReentrantLock;
+import java.util.function.Consumer;
+
+public final class Acquisition {
+
+    private static final ThreadLocal<ScheduledAcquisition> SCHEDULED_ACQUISITION = ThreadLocal.withInitial(ScheduledAcquisition::new);
+
+    /**
+     * Global lock used for synchronization.
+     */
+    private static final ReentrantLock GLOBAL_LOCK = new ReentrantLock();
+
+    private static final AtomicLong WAIT_COUNTER_NANO = new AtomicLong();
+
+    /**
+     * Acquires a {@link Collection}.
+     * <p>
+     * Order is not guaranteed.
+     *
+     * @param collection the collection to acquire
+     * @param consumer   the consumer called for each of the collection element
+     * @param <E>        the object type
+     */
+    public static <E> void acquireForEach(@NotNull Collection<Acquirable<E>> collection,
```

Comment:
I wonder if this could take a consumer on a collection of E so that it could be grouped together. For example, the retrieve thread map method could collect the ones in the current thread and run them all at once. This method can still exist and overload the other one if needs be.